_posts/2020-09-06-how-the-system-tests-works.md

```diff
@@ -0,0 +1,596 @@
+
+####  1. Kafka with new persistent claim storage and following settings
+
+##### <a id="javabeans">Code snippet 1.1 Using JavaBeans convention</a>
+
+```java
+PersistentClaimStorage psc = new PersistentClaimStorage();
+
+// JavaBeanss convention having setters 
+psc.setId(0);
+psc.setSize("100Gi");
+psc.setDeleteClaim(true);
+
+KafkaResource.kafkaEphemeral(CLUSTER_NAME, 3, 1)
+    .editSpec()
+        .editKafka()
+            .withPersistentClaimStorage(psc)
+        .endKafka()
+    .endSpec()
+    .done();
+```
+
+##### <a id="builder">Code snippet 1.2 Using Builder pattern</a>
+```java
+// Builder pattern by which we can create one type of object
+PersistentClaimStorage pcs = new PersistentClaimStorageBuilder()
+    .withDeleteClaim(true)
+    .withId(0)
+    .withSize("100Gi")
+    .build();
+
+KafkaResource.kafkaEphemeral(CLUSTER_NAME, 3, 1)
+    .editSpec()
+        .editKafka()
+            .withPersistentClaimStorage(psc)
+        .endKafka()
+    .endSpec()
+    .done();
+```
+
+##### <a id="abstractbuilder">Code snippet 1.3 Using Abstract recursive builder</a>
+```java
+// Abstract recursive builder which can have more objects in hierarchy as you can see here
+KafkaResource.kafkaEphemeral(CLUSTER_NAME, 3, 1)
+    .editSpec()
+        .editKafka()
+            .withNewPersistentClaimStorage()
+                .withId(0)
+                .withNewSize("100Gi")
+                .withDeleteClaim(true)
+            .endPersistentClaimStorage()
+        .endKafka()
+    .endSpec()
+    .done();
+```
+
+This examples was illustrated idea only one dependent you can imagine three, four and more objects, which can be changed for instance:
+
+```
+KafkaResource.kafkaEphemeral(CLUSTER_NAME, 3, 1)
+    .editMetadata()
+        .withName(name)
+        .withNamespace(ResourceManager.kubeClient().getNamespace())
+    .endMetadata()
+    .editSpec()
+        .editKafka()
+            .withVersion(Environment.ST_KAFKA_VERSION)
+            .withReplicas(kafkaReplicas)
+            .addToConfig("log.message.format.version", TestKafkaVersion.getKafkaVersionsInMap().get(Environment.ST_KAFKA_VERSION).protocolVersion())
+            .addToConfig("offsets.topic.replication.factor", Math.min(kafkaReplicas, 3))
+            .addToConfig("transaction.state.log.min.isr", Math.min(kafkaReplicas, 2))
+            .addToConfig("transaction.state.log.replication.factor", Math.min(kafkaReplicas, 3))
+            .withNewListeners()
+                .withNewPlain().endPlain()
+                .withNewTls().endTls()
+            .endListeners()
+            .withNewInlineLogging()
+                .addToLoggers("log4j.rootLogger", "DEBUG")
+            .endInlineLogging()
+        .endKafka()
+        ...
+        ... (more configs...)
+        ... 
+    .endSpec()
+    .done()
+```
+
+In this case you can imagine that it is really easy to create multiple different instances of the `Kafka` instead of having separate objects.
```

Comment:
```suggestion
In this case you can see that it is really easy to create multiple different instances of the `Kafka` instead of having separate objects.
```